homeassistant/components/sensibo/config_flow.py

```diff
@@ -24,6 +26,52 @@ class SensiboConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     VERSION = 2
 
+    entry: config_entries.ConfigEntry | None
+
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle re-authentication with Sensibo."""
+
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm re-authentication with Sensibo."""
+        errors: dict[str, str] = {}
+
+        if user_input:
+            api_key = user_input[CONF_API_KEY]
+            try:
+                await async_validate_api(self.hass, api_key)
```

Comment:
We still need to validate the username is the same

Reply:
Good catch! Completely missed that it could of course be a different account.